### PR TITLE
Trim TCX files before importing

### PR DESCRIPTION
**Pull request by Danysan1**
TLDR: Allows to import TCX files with leading whitespaces, such as those generated by Strava.

As per XML standard the XML declaration must be the first node in the document, and no white space characters are allowed to appear before it.

Unfortunately some companies do not always respect this rule, amongst them Strava. This makes importing files exported from them difficult. This problem has already been reported in the past in #3149 but it was not addressed. Stumbling upon this problem myself I have tried to fix it.

Except from trimming leading whitespaces this commit has no other intended functional changes. In my tests this holds true.

### CODE REVIEW

**Comment by Danysan1** [comment]
I don't know why the Travis CI Linux build is failing. When building with QtCreator on my linux machine it compiles correctly. The Travis build ends with 

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
```

Maybe it's just a fluke in Travis CI?

**Comment by amtriathlon** [comment]
> Maybe it's just a fluke in Travis CI?

Yes
